### DISCUSSION TITLE
Request a configuration to limit the time and number for generating data

### DISCUSSION THREAD

**Discussion by AndyLuo1029** [synthetichealth/synthea#1396] [Q&A]
I am trying to use Synthea to generate a small sample data set that is **easy to extract data**. My ideal data set should be able to accurately limit the number of records of each type of data. **The purpose is to easily and quickly map JSON in FHIR format and text in Text format.** Taking text format data as an example, modules such as ALLERGIES and OBSERVATIONS should have only one/several records. I hope to achieve this goal by controlling the specific date of the record or by controlling the number of records, but the Synthea configuration method currently given in the wiki cannot meet my requirements.

Is there any way to achieve my goal in the current version of Synthea? If not, is there a way to quickly correspond to FHIR's JSON format and Text format? 

Many thanks for all answers and help!

**Comment by StevenMAustin** [comment]
Hello Luo, 
You might want to check out app.makedata.ai to get quick and easy access to Synthea data. As a guest user, you can request up to 10 records. You can download selected records or all results to JSON, C-CDA, or CSV. When you request records you can see right away which ones have Allergies or Observations. I hope this is helpful!

**Comment by jawalonoski** [comment]
You _can_ specify the number of patients you want, the years of history to export (default is 10, but if you decrease that number the amount of data will decrease), and export formats.

You _cannot_ specify the quantity of individual export data... For instance, you cannot specify the number of FHIR `Observations` or how many rows appear in the `procedures.csv` file.

My recommendation is to vary the number of patients (`-p`) and limit the years of history (`--exporter.years_of_history=3`).
